src/Client/Pages/Catalog/Brands.razor.cs

```diff
@@ -43,8 +43,11 @@ private async Task<TableData<BrandDto>> ServerReload(TableState state)
             state.Page = 0;
         }
 
-        await LoadDataAsync(state);
-
+        if (!_loading)
```

Comment:
better to put the check inside LoadDataAsync probably? It's now already on another place in this file (OnSearch), and it's not even big yet ;-)